File: backend/src/main/java/com/kevinlemein/backend/model/Appointment.java
package com.kevinlemein.backend.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "appointments")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder

public class Appointment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "patient_id" , nullable = false)
    private Patient patient;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "doctor_id" , nullable = false)
    private User doctor;

    @Column (nullable = false)
    private LocalDateTime appointmentDateTime;

    @Enumerated (EnumType.STRING)
    @Column(nullable = false)
    private AppointmentStatus appointmentStatus;

    @Column(columnDefinition = "TEXT")
    private String reason;

    @Column(columnDefinition = "TEXT")
    private String notes;

    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        this.createdAt = LocalDateTime.now();
        this.updatedAt = LocalDateTime.now();

        if(appointmentStatus == null){
            appointmentStatus = AppointmentStatus.PENDING;
        }
    }

    @PreUpdate
    protected void onUpdate() {
        this.updatedAt = LocalDateTime.now();
    }

}

File: dotnetapi/Models/Appointment.cs
using System.Text.Json.Serialization;

namespace Business.Entities
{
    public class Appointment
    {
        public long Id { get; set; }
        public DateTime AppointmentDateTime { get; set; }
        public int? StatusId { get; set; }
        public string Status { get; set; }
        public DateTime CreatedAt { get; set; }
        public string? Notes { get; set; }       // Nullable since text can be null
        public string? Reason { get; set; }      // Nullable since text can be null
        public DateTime? UpdatedAt { get; set; } // Nullable
        public long DoctorId { get; set; }
        public Doctor Doctor { get; set; }
        public long PatientId { get; set; }
        public Patient Patient { get; set; }
    }

    public class AppointmentStatus : CreateFields
    {
        public int Id { get; set; }
        public string Name { get; set; }
    }
    public class Prescription
    {
        public long Id { get; set; }
        public long? AppointmentId { get; set; }
        [JsonIgnore]
        public Appointment Appointment { get; set; }
        
        public int? DrugId { get; set; }
        [JsonIgnore]
        public Drug Drug { get; set; }
        public string DrugName =>Drug?.Name;
        public int Dossage { get; set; }
        public string Duration { get; set; }
        
        public DateTime CreatedOn { get; set; }
        public long DoctorId { get; set; }
        [JsonIgnore]
        public Doctor Doctor { get; set; }
        public string Notes { get; set; }
    }
    public class MedicalRecord : CreateFields
    {
        public long Id { get; set; }
        public long? AppointmentId { get; set; }
        [JsonIgnore]
        public Appointment Appointment { get; set; }
        public string Diagnosis { get; set; }
  
    }
}
